daemon/cluster/cluster.go

```diff
@@ -345,11 +344,19 @@ func (c *Cluster) Cleanup() {
 			}
 		}
 	}
+
+	// Release the lock before stopping the node, so that calls back into
+	// Cluster during the shutdown process do not hang.
+	c.mu.Unlock()
```

Comment:
Nothing in the block above seems to use `c.mu` so this could be combined with the other unlock in line 334